code/game/objects/structures/barricade.dm

```diff
@@ -277,6 +277,14 @@
 	setDir(turn(dir, 270))
 
 
+/obj/structure/barricade/attack_hand_alternate(mob/living/user)
+	if(anchored)
+		to_chat(user, "<span class='warning'>It is fastened to the floor, you can't rotate it!</span>")
+		return FALSE
+
+	setDir(turn(dir, 270))
```

Comment:
Might want to make a single proc used for all rotate, rather than copypasting

Reply:
then you're just copypasting that rotate ? thonk